payment-circle/src/main/java/org/stellar/anchor/paymentservice/circle/StellarReconciliation.java

```diff
@@ -0,0 +1,79 @@
+package org.stellar.anchor.paymentservice.circle;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.List;
+import org.stellar.anchor.exception.HttpException;
+import org.stellar.anchor.paymentservice.circle.model.CircleTransactionParty;
+import org.stellar.anchor.paymentservice.circle.model.CircleTransfer;
+import org.stellar.sdk.Server;
+import org.stellar.sdk.responses.Page;
+import org.stellar.sdk.responses.operations.OperationResponse;
+import org.stellar.sdk.responses.operations.PathPaymentBaseOperationResponse;
+import org.stellar.sdk.responses.operations.PaymentOperationResponse;
+import reactor.core.publisher.Mono;
+
+public interface StellarReconciliation {
+
+  Server getHorizonServer();
+
+  default void updateStellarSenderAddress(CircleTransfer transfer) throws HttpException {
+    if (transfer.getSource().getId() != null) return;
+
+    // Only Stellar->CircleWallet transfers would arrive here with id == null, and they'd also have
+    // chain == "XLM" and type == BLOCKCHAIN
+    if (!transfer.getSource().getChain().equals("XLM")
+        || transfer.getSource().getType() != CircleTransactionParty.Type.BLOCKCHAIN) {
+      throw new HttpException(500, "invalid source account");
+    }
+
+    Page<OperationResponse> responsePage;
+    try {
+      String txHash = transfer.getTransactionHash();
+      responsePage = getHorizonServer().payments().forTransaction(txHash).execute();
```

Comment:
This is a blocking IO call, shouldn't we try to do this using our web client?

Reply:
Oh I see, I wasn't aware of this mechanic. At the base layer though, using the SDK's horizon server will block a thread right? Even if its not the main thread?

Reply:
Yes, that's correct.